src/extended_json.ts

```diff
@@ -172,65 +192,67 @@ const BSON_INT32_MAX = 0x7fffffff,
  *
  * // prints '{"int32":10}'
  * console.log(EJSON.stringify(doc));
+ * ```
  */
+export function stringify(value: BSONDocument): string;
+export function stringify(value: BSONDocument, options?: EJSONOptions): string;
 export function stringify(
-  value: any,
-  replacer?: (this: any, key: string, value: any) => any,
+  value: BSONDocument,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  replacer?: (number | string)[] | ((this: any, key: string, value: any) => any) | EJSONOptions,
```

Comment:
The only reason it can't is that it would be unpleasant for users, their callbacks which might be inlined declarations would have to do type assertions
```js
EJSON.stringify(doc, (k, v) => {
  return v.toFixed(2); // maybe I know my values are numbers but I can't just do this freely anymore if we use unknown
})
```
It would also add a type incompatibility with JSON.stringify, so you couldn't share replacers.